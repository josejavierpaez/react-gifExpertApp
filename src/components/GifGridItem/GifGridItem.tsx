import React from 'react'
import { IGifGridItemProps } from '../../interfaces/interfaces'
export const GifGridItem = ({ id, title, url }: IGifGridItemProps) => {

  return (
    <div className="card animate__animated animate__pulse">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}
