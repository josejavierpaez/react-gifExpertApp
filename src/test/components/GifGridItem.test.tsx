import { GifGridItem } from '../../components/GifGridItem/GifGridItem';
import { shallow } from 'enzyme';

describe('test to GifGridItem component', () => {

  const title = 'title';
  const url = 'https://localhost/algo.jpg';
  const wraper = shallow(<GifGridItem title={title} url={url} />);

  test('should be render GifGridItem component well', () => {

    expect(wraper).toMatchSnapshot();
  })

  test('should be have p tag with the title', () => {
    const p = wraper.find('p');
    expect(p.text().trim()).toBe(title);
  })

  test('should be have img equal to props url and alt', () => {
    const img = wraper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })

  test('should have animate__pulse', () => {
    const div = wraper.find('div');
    expect(div.hasClass('animate__pulse')).toBe(true);
  })

});
