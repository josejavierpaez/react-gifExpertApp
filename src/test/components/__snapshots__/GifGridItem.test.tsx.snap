// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to GifGridItem component should be render GifGridItem component well 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="title"
    src="https://localhost/algo.jpg"
  />
  <p>
    title
  </p>
</div>
`;
